### ISSUE TITLE
Dimension setting missing in host.name in system package

### ISSUE THREAD

**Issue by ishleenk17** [elastic/integrations#8256]
host.name is not specified as dimension field in some of the datastreams in System Package.

Below datastreams have this setting missing in ecs.yml: 

[Diskio Datastream](https://github.com/elastic/integrations/blob/main/packages/system/data_stream/diskio/fields/ecs.yml)
[Memory Datastream](https://github.com/elastic/integrations/blob/main/packages/system/data_stream/memory/fields/ecs.yml)

PR to fix the issue - https://github.com/elastic/integrations/pull/8261

**Comment by lalit-satapathy** [comment]
Looks like _host.name_ as got removed from _agent.yml_ in this PR - https://github.com/elastic/integrations/pull/7006/ ?

@tetianakravchenko, Can you confirm where is _host.name_ as dimension modeled in this data stream currently?

CC: @agithomas @andrewkroh

**Comment by ishleenk17** [comment]
For memory it got removed as part of this [cleanup PR](https://github.com/elastic/integrations/pull/8100), for diskio as part of #7006

**Comment by tetianakravchenko** [comment]
@lalit-satapathy 

> @tetianakravchenko, Can you confirm where is host.name as dimension modeled in this diskio stream currently?

yes, It should be added, my mistake. @ishleenk17 thank you for adding it back!

**Comment by andrewkroh** [comment]
> For memory it got removed as part of this https://github.com/elastic/integrations/pull/8100, for diskio as part of https://github.com/elastic/integrations/pull/7006

#8100 did not remove any `dimension` fields according to the diff. It was performed by a tool that [was designed](https://github.com/andrewkroh/fydler/blob/eddf66dada2beed9a1006b5c33f812fd56788658/internal/analysis/useecs/useecs.go#L122-L124) to retain `dimension` while converting to `external: ecs`.

7006 shows a removal at https://github.com/elastic/integrations/commit/8040757ce8aa9471554ae8c62620d2bf52fe6da7#diff-eae9e9ef0ee713b08ca5d100764c8ceeb01066b585d3f1971cc5b3470ef11979L142

**Comment by ishleenk17** [comment]
> #8100 did not remove any `dimension` fields according to the diff. It was performed by a tool that [was designed](https://github.com/andrewkroh/fydler/blob/eddf66dada2beed9a1006b5c33f812fd56788658/internal/analysis/useecs/useecs.go#L122-L124) to retain `dimension` while converting to `external: ecs`.

@andrewkroh : I suppose this tool has added the dimension field to host.id in [agent.yml](https://github.com/andrewkroh/integrations/blob/44284b68d34a51558e6a5d761f898509e210c097/packages/system/data_stream/memory/fields/agent.yml#L49) and removed it from host.name, since host.name was present in ecs.yml

**Comment by andrewkroh** [comment]
@tetianakravchenko Thanks. I see where I went wrong. It wasn't the tooling. It was the manual edit in https://github.com/elastic/integrations/pull/8100/commits/8e0b974f25ebe5375e4c3d64fb32c89534160a43#diff-0b3481494795026f41db4bdcf201a3ce961c3eda2dda10c5cd582f7eca01e8baL55-L56 where I had to decide which duplicate field definition of the `host.name` field to remove, and I mistakenly removed the one with the `dimension` attribute instead of the one without it.
